planner/optimize.go

```diff
@@ -38,12 +39,21 @@ import (
 // Optimize does optimization and creates a Plan.
 // The node must be prepared first.
 func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is infoschema.InfoSchema) (plannercore.Plan, types.NameSlice, error) {
-	fp := plannercore.TryFastPlan(sctx, node)
-	if fp != nil {
-		if !useMaxTS(sctx, fp) {
-			sctx.PrepareTSFuture(ctx)
+	_, isolationReadContainTiKV := sctx.GetSessionVars().GetIsolationReadEngines()[kv.TiKV]
+	cfgIsolationReadContainTiKV := false
+	for _, engine := range config.GetGlobalConfig().IsolationRead.Engines {
+		if engine == kv.TiKV.Name() {
+			cfgIsolationReadContainTiKV = true
+		}
+	}
```

Comment:
Those lines can be skipped if `isolationReadContainTiKV` is false.

Reply:
It's unnecessary, the size of engines is less than three. If we check `isolationReadContainTiKV` first, it makes the code more complex.

Reply:
It can improve some performance, especially in heavy OLTP workloads. PTAL